### DISCUSSION TITLE
How to setup and timer attach it to a function and Start / Stop it

### DISCUSSION THREAD

**Discussion by rowifi** [contrem/arduino-timer#49] [Q&A]
I've made timers that repeatedly call a function, but I want to set up the timer, and at some point Stop it., then start it elsewhere.
Or alternatively setup a timer, attach a function and have it start later in the code not at the time I set up it.
I can't see the correct procedures to do this.

**Comment by philj404** [comment]
It sounds like you want something like a "pause" to temporarily skip a task.  That is not supported directly, But you can get equivalent behavior.

The most direct way is to use `timer.cancel(myTimerTaskId)` to cancel your task, then call `timer.every(...)` again once it is time to restart it.

Another way is to modify your task to always run, but skip your 'interesting' code when the task  is 'paused':
```
bool pauseBlinking = true;

bool myBlink(void) {
    if(pauseBlinking ) {
        return true;  // no blinking for now, but keep the task active/ready
    }
    // task code -- time to do something interesting
    ...
}
...
    myTimerTaskId = timer.every(500, myBlink);
...
    pauseBlinking = false;  // blink for now
...
    pauseBlinking = true;  // stop blinking
...
```
Then some code sets `pauseBlinking` to `true` or `false` depending on whether you want your code to blink.
